### ISSUE TITLE
RestTemplate consumes XML instead of JSON with Brixton.BUILD-SNAPSHOT 

### ISSUE THREAD

**Issue by snussbaumer** [spring-cloud/spring-cloud-netflix#852]
`spring-cloud-starter-eureka.BUILD-SNAPSHOT` pulls the `eureka-core.1.4.1` dependency which in turns pulls `jackson-dataformat-xml.2.6.5`
When jackons-dataformat-xml is on the classpath, RestTemplate adds application/xml first in the `Accept` header. When calling an endpoint that can produce XML as well as JSON, you now get XML in return when you would get JSON with Brixton.M4.

The problem is that the Xml parser (`FromXmlParser`) is not as good as the JSON parser (`ReaderBasedJsonParser`), for example it will deserialize boolean values as Strings in maps.

Below is an integration test that fails with Brixton.BUILD-SNAPSHOT but passes with Brixton.M4 : 

```java
@RunWith(SpringJUnit4ClassRunner.class)
@SpringApplicationConfiguration(classes = IssueReproApplication.class)
@WebIntegrationTest(randomPort = true)
public class RestTemplateIT {

    @Value("${local.server.port}")
    private int port;

    @Test
    public void restTemplateIssue() throws Exception {
        RestTemplate restTemplate = new RestTemplate();
        ResponseEntity<StringObjectMap> result = restTemplate.exchange(new URI("http://localhost:" + port + "/issue"),
                HttpMethod.GET, null, StringObjectMap.class);
        assertThat(result).isNotNull();
        assertThat(result.getStatusCode()).isEqualTo(HttpStatus.OK);
        assertThat(result.getBody()).containsEntry("active", true);
    }

    @SuppressWarnings("serial")
    public static class StringObjectMap extends HashMap<String, Object> {
    }

    @RestController
    public static class Controller {
        @RequestMapping(path = "/issue")
        public StringObjectMap issue() {
            StringObjectMap result = new StringObjectMap();
            result.put("active", true);
            return result;
        }
    }

    @SpringBootApplication
    public static class IssueReproApplication {
        public static void main(String... args) {
            SpringApplication.run(IssueReproApplication.class, args);
        }

        @Bean
        @Primary
        public DropwizardMetricServices dropwizardMetricServices(MetricRegistry metricRegistry) {
            return new DropwizardMetricServices(metricRegistry);
        }
    }
}
```

The error I get is : 
```
Expecting:
 <{"active"="true"}>
to contain:
 <[MapEntry[key="active", value=true]]>
but could not find:
 <[MapEntry[key="active", value=true]]>
```

The difference is that the boolean value is converted to the string `"true"` instead of the boolean `true`

You can workaround the problem by either adding a `produces = MediaType.APPLICATION_JSON_UTF8_VALUE` in the `RequestMapping` to force JSON on the server side, or ask for JSON on the client side by setting the `Accept` header to `application/json` in the restTemplate request entity.

This is not showstopper ... but still a bit sneaky as a  spring based microservice system relying on the defaults value everywhere can switch overnight from JSON to XML as the message format, maybe without even noticing...

**Comment by spencergibb** [comment]
@snussbaumer I believe this was fixed in bc0eef9ae2c2f300054d7c0baf22091c184e0c69, can you confirm?

**Comment by snussbaumer** [comment]
@spencergibb Yes I confirm this is fixed in BUILD-SNAPSHOT this morning. Thanks !
